apps/comps/rpc/middleware/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { StandardRPCJsonSerializer } from "@orpc/client/standard";
+import {
+  Context,
+  ErrorMap,
+  Meta,
+  Middleware,
+  ORPCErrorConstructorMap,
+  os,
+} from "@orpc/server";
+import { unstable_cache } from "next/cache";
+import type { Logger } from "pino";
+
+export function cacheMiddleware<
+  TInContext extends Context & { logger: Logger },
+  TInput,
+  TOutput,
+  TErrorMap extends ErrorMap,
+  TMeta extends Meta,
+>(options?: {
+  key?: string[];
+  revalidate?: number;
+  tags?: string[];
```

Comment:
These args mirror the args to the NextJS `unstable_cache` function and will be passed through to it.